### ISSUE TITLE
Cutting a color painted object removes all paint

### ISSUE THREAD

**Issue by alexzogh** [bambulab/BambuStudio#750]
**Describe the bug**

I attempted to cut a painted model that was too big for the plate. As soon as I cut the model, all of the paint disappeared on the colored model. 

**3mf File for This Bug**

**To Reproduce**
1. choose any painted object in the 3mf
2. Click on the cut icon
3. click on 'perform cut'
4. See error - the object will now be cut, but naked with no paint

**Expected behavior**

I expected the object to be cut, but all of the paint to be intact

**Screenshots**

![colored1](https://user-images.githubusercontent.com/16065535/205191812-fa9e2aac-7f90-4f72-b047-db54bbadb022.jpg)
Photo of a selected colored object about to be cut

![colored2](https://user-images.githubusercontent.com/16065535/205191864-029ce891-cfd8-44a3-9c9d-5f0ded8553db.JPG)
Photo of the same selected colored object after clicking "perform cut" - all painting has been removed

**Desktop (please complete the following information):**
 - OS: Windows 10 version 22H2
 - Bambu Studio version 01.03.01.01

**Comment by lanewei120** [comment]
it is the current behavior by design,  not a bug
marked as a feature

**Comment by randyre** [comment]
I'm beginning to wonder why so much of the basic functionality of Prusia Slicer has been cut out in the first place. It seems harder to add all this back in than to just make PS work as is with an X1.

**Comment by YifanWuBambu** [comment]
@alexzogh @randyre 
This multi-color model is not generated by the multimaterial painting tool. It's color is assigned by specify the extruder of each part of object in the object list or context menu. Bambu Studio also support this.
![paint_test_1](https://user-images.githubusercontent.com/110661856/205419940-e8ddf1bf-23a5-4f9d-abcd-608ca357b0c3.gif)

**Comment by ResiduaLiu** [comment]
> How in the world is undoing and fundamentally changing everything about an object when you just want it cut a 'feature'? Who in the world would ask for such a feature? When I resize the object I don't lose the painting. When I Split, or modify (add, subtract, etc..) it doesn't lose the painting. So are those all bugs then? Very inconsistent behavior when some features behave one way, and others don't.

You are right, but I think you misunderstand the reply. "Marked as feature" means this github issue will be labeled as "feature" rather than "bug". It is a reference for the developers.  The Bambu team will solve the problem you proposed.

**Comment by rafffel** [comment]
Any news about that one ?

**Comment by randyre** [comment]
@YifanWuBambu
Is there a way to have it only affect the triangles that touch the cut and put them back to the base color, and leave the rest of the model colored, That way there's not as much work to do to get painted again.

**Comment by bachrach44** [comment]
Any update on this? I also find it super annoying to spend all this time coloring an object, only to have all my hard work cancelled when I try to set it up for printing.

**Comment by remcoder** [comment]
I just ran into this in v1.8.2.56 too. It's a limitation that's easily encountered.

btw, if you lose the colors accidentally this way. At least know you can CTRL-Z to undo the cut.

**Comment by partymandkse** [comment]
I guess this one have been going one for a while, but i'd like to give my +1 to this annoying issue.
Haven't tried Orca slicer yet, but this certainly makes me wannna do it.

**Comment by pgifford** [comment]
> I guess this one have been going one for a while, but i'd like to give my +1 to this annoying issue. Haven't tried Orca slicer yet, but this certainly makes me wannna do it.

OrcaSlicer has the same problem which is not surprising since it is periodically updated to the latest Bambu Studio code.

**Comment by cew214DAL** [comment]
Adding my +1 as well.  Just ran into the issue when just trying to remove the non-colored edges of a file.  Not touching any of the colored triangles at all.  If making this change is so difficult, how about a feature where you can convert painted objects to actual objects (apologies if that's already a thing and I just missed it).  That way, I can just drag the painted object to another plate, do my cut and resize the painted object(s) to whatever size I need and drag it back to the original plate.

**Comment by CallmeBigfat** [comment]
still a thing which is crazy since bambu is the most popular printer company now days.

**Comment by Timefactor** [comment]
Could we get an update on this issue? Just purchased an X1 Carbon and AMS. Spent time painting a very large shield for a cosplay event and when I went to cut it so it could fit on the 256x256x256 plate, all the painting disappeared. This bug has been open for 3 years now. Doesn't help me gain confidence with the brand when the dev team takes over 3 years to resolve an issue like this.

**Comment by alexzogh** [comment]
I was the original author of the bug almost 3 years ago. There is a workflow workaround that works just fine for me, and, once you know it, this won't be an issue going forward.

First, Make your cuts FIRST. Most important, when you make your cuts, make sure you check the box next to "cut to parts"

<img width="263" height="326" alt="Image" src="https://github.com/user-attachments/assets/3aeaa8ec-7fc1-49ee-8898-91016fee0b6b" />

When you are done cutting all of your parts, go back to the main section, and click on the icon for "assembly view"

<img width="113" height="97" alt="Image" src="https://github.com/user-attachments/assets/aca74a89-9b49-44f0-999f-f96ba23790cd" />

This will assemble all of the parts into one object, and you can now paint the object when it is assembled, and the paint will be back onto the individual parts.

<img width="970" height="532" alt="Image" src="https://github.com/user-attachments/assets/928017f3-ab87-4272-8c42-883989ccd92c" />

Not ideal, but since they added assemblies, I've been doing it this way, and it's worked out well.

**Comment by CallmeBigfat** [comment]
> Could we get an update on this issue? Just purchased an X1 Carbon and AMS. Spent time painting a very large shield for a cosplay event and when I went to cut it so it could fit on the 256x256x256 plate, all the painting disappeared. This bug has been open for 3 years now. Doesn't help me gain confidence with the brand when the dev team takes over 3 years to resolve an issue like this.

I have another bug they have not fixed in 3 years so I agree 100%. Its a shame. I have been just putting a negative "slice" for the objects I have worked with after coloring. But I know this is not a solution for everything. I will need to try ops solution above next :)

**Comment by solidblu** [comment]
I have a Patch I am testing for this that  I am working on.  I tested that it works in Windows only so far, still need to build and test on Mac and Linux before I actually submitting.  However if anyone who really wants to try it out and has done local builds before:

https://github.com/solidblu/BambuStudio/pull/1/changes  Based on master from commit:
dacd7881118089e237f22b0d7afb0fe4c728ee49

<img width="887" height="590" alt="Image" src="https://github.com/user-attachments/assets/f65a3a11-e8b7-4146-8ab2-4e8c3cf94051" />
<img width="1775" height="1186" alt="Image" src="https://github.com/user-attachments/assets/e61153b0-6ddf-4d27-9ea7-a396ae284bdc" />
<img width="1769" height="1182" alt="Image" src="https://github.com/user-attachments/assets/accab42c-f1cf-4316-b646-67b2332304fd" />

**Comment by solidblu** [comment]
Created PR to fix this. If it is to the wrong branch or anything happy to change / close.  Just trying to help community.

**Comment by Cstm3DBldr** [comment]
For anyone landing here: paint loss on **Cut** is already fixed as of BambuStudio 2.7.1 (via #10613). I've opened **PR #11105**, which extends the same paint-preservation to the operations that *still* drop it — **merge, split, convert to/from multi-part, and Repair**. Same underlying problem (painted face data not carried through model edits), so linking it here.
